SC_redesign/Crop_and_Soil/field/tillage_application.py

```diff
@@ -0,0 +1,199 @@
+        ----------
+        SWAT Theoretical documentation, example on page 361.
+
+        Notes
+        -----
+        This method executes the actual mixing between the soil layers. Each layer in the soil profile can be either
+        fully tilled, partially tilled, or not tilled at all. The method starts by determining how much total stuff will
+        be mixed in the profile based on the mixing fraction and how much stuff is in the pool of each layer. Then it
+        redistributes mixed stuff back into the tilled layers of the profile. The amount of stuff mixed back in to a
+        layer is determined by the ratio between the depth of tillage in the layer and the total overall tillage depth.
+
+        """
+        redistribution_fractions = []
+        total_to_mix_from_pools = 0
+        for layer in self.soil_data.soil_layers:
+            layer_not_tilled = layer.top_depth >= tillage_depth
+            layer_partially_tilled = layer.bottom_depth > tillage_depth
+            if layer_not_tilled:
+                break
+            elif layer_partially_tilled:
+                tilled_depth = tillage_depth - layer.top_depth
+                layer_redistribution_fraction = tilled_depth / tillage_depth
+                fraction_of_layer_mixed = tilled_depth / layer.layer_thickness
+            else:
+                layer_redistribution_fraction = layer.layer_thickness / tillage_depth
+                fraction_of_layer_mixed = 1.0
+            redistribution_fractions.append(layer_redistribution_fraction)
+
+            current_pool_amount = getattr(layer, pool_name)
+            amount_to_remove = current_pool_amount * mixing_fraction * fraction_of_layer_mixed
+            unmixed_amount_in_pool = current_pool_amount - amount_to_remove
+            setattr(layer, pool_name, unmixed_amount_in_pool)
+            total_to_mix_from_pools += amount_to_remove
+
+        number_of_tilled_layers = len(redistribution_fractions)
+        for layer_index in range(number_of_tilled_layers):
+            layer = self.soil_data.soil_layers[layer_index]
+            layer_fraction = redistribution_fractions[layer_index]
+
+            amount_to_add = total_to_mix_from_pools * layer_fraction
+
+            amount_in_pool = getattr(layer, pool_name)
+            new_pool_amount = amount_in_pool + amount_to_add
+            setattr(layer, pool_name, new_pool_amount)
+
+    @staticmethod
+    def _remove_amount_incorporated(data_container: object, attribute_name: str,
+                                    incorporation_fraction: float) -> float:
+        """
+        Calculates amount of stuff incorporated from the soil surface into the soil profile.
+
+        Parameters
+        ----------
+        data_container : object
+            Class instance containing the soil surface pool to be removed from (unitless)
+        attribute_name : str
+            Name of the pool to be removed from (unitless)
+        incorporation_fraction : float
```

Comment:
```suggestion
        data_container : object
            Class instance containing the soil surface pool to be removed from 
        attribute_name : str
            Name of the pool to be removed from
        incorporation_fraction : float
```
I don't think they're kind of redundant in this use case.